File: LowLevelDesign/java/task-scheduler/src/com/taskscheduler/task/PrintMessageTask.java
package com.taskscheduler.task;

import java.time.LocalDateTime;

public class PrintMessageTask implements Task {
    private final String message;

    public PrintMessageTask(String message) {
        this.message = message;
    }

    @Override
    public void execute() {
        System.out.printf("[%s] Executing PrintMessageTask: %s%n", LocalDateTime.now(), message);
    }
}

File: LowLevelDesign/task-scheduler-nodejs/src/task/PrintMessageTask.js
import { Task } from './Task.js';

/**
 * PrintMessageTask - Example task implementation that prints a message
 */
export class PrintMessageTask extends Task {
    constructor(message) {
        super();
        this.message = message;
    }

    async execute() {
        console.log(`[${new Date().toISOString()}] Executing PrintMessageTask: ${this.message}`);
    }
}
